javascript/javascript1/week3/index.js

```diff
@@ -0,0 +1,244 @@
+const names = [
+    "Peter",
+    "Ahmad",
+    "Yana",
+    "kristina",
+    "Rasmus",
+    "Samuel",
+    "katrine",
+    "Tala",
+];
+const nameToRemove = "Ahmad";
+for (let i = 0; i < names.length; i++) {
+    if (names[i] === nameToRemove) {
+        names.splice(i, 1)
+    }
+}
+console.log(names); // ['Peter', 'Yana', 'kristina', 'Rasmus', 'Samuel', 'katrine', 'Tala']
+
+
+//question 2: 8 hours and 38 minutes
+const travelInformation = {
+    speed: 50,
+    destinationDistance: 432,
+};
+const travelTime = getTravelTime(travelInformation);
+function getTravelTime() {
+    let result = travelInformation.destinationDistance / travelInformation.speed; //8.64
+    return Math.floor(result) + " hours and " + Math.floor((result - Math.floor(result)) * 60) + " minutes.";
+}
+console.log(travelTime);
+
+//question 3:Series duration of my life---------
+const seriesDurations = [
+    {
+        title: "Run, brothers",
+        days: 27,
+        hours: 12,
+        minutes: 24,
+    },
+    {
+        title: "Movie&TV Awards",
+        days: 43,
+        hours: 10,
+        minutes: 24,
+    },
+    {
+        title: "MTV Videos Music Awards",
+        days: 46,
+        hours: 14,
+        minutes: 24,
+    },
+    {
+        title: "Game of thrones",
+        days: 3,
+        hours: 1,
+        minutes: 0,
+    }
+];
+
+function logOutSeriesText(seriesDurations) {
+    // write code here
+    let string = "";
+    let percent = 0;
+    for (const newSeries of seriesDurations) {
+        //count the percent of each series
+        percent = (newSeries.days * 24 * 60 + newSeries.hours * 60 + newSeries.minutes) * 100 / (80 * 365 * 24 * 60);
+        //out put percent
+        string = string + newSeries.title + " took " + Math.floor(percent * 100) / 100 + " % of my life\n\n";
+    }
+
+    console.log(string);
+}
+
+logOutSeriesText(seriesDurations);
+
+// this is the example below--------
+const seriesDurations1 = [
+    {
+        title: "Game of thrones",
+        days: 3,
+        hours: 1,
+        minutes: 0,
+    }
+];
+
+function logOutSeriesText1(seriesDurations) {
+    // write code here
+    const percent = (seriesDurations1[0].days * 24 * 60 + seriesDurations1[0].hours * 60 + seriesDurations1[0].minutes) * 100 / (80 * 365 * 24 * 60);
+
+    return seriesDurations1[0].title + " took " + Math.floor(percent * 100) / 100 + " % of my life ";
+}
+
+console.log(logOutSeriesText1(seriesDurations1)); // logs out the text found above
+
+
+//question 4:NOnoN0nOYes (Note taking app)-------------
+
+const notes = [];
+// save a note------------------------
+function saveNote(content, id) {
+    notes.push({
+        "content": content,
+        "id": id
+    })
+}
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes); // [{content: 'Pick up groceries', id: 1}, {content: 'Do laundry', id: 2}]
+
+
+//Get a note------------
+function getNote(id) {
+    // get the object with name of id
+    for (const getId of notes) {
```

Comment:
I think note or Anote is a better naming than getId.It's better to start function naming with the verb. you could check this link for more info:
[https://javascript.plainenglish.io/javascript-naming-convention-best-practices-b2065694b7d](url)

Reply:
I can not open the link.